packages/ui/src/proposals/components/ProposalStages.tsx

```diff
@@ -3,44 +3,73 @@ import React, { useMemo } from 'react'
-  const rounds = useMemo(() => {
+type RoundState = 'approved' | 'rejected' | 'deciding' | 'disabled'
+const iconMap = { approved: <CheckboxIcon />, rejected: <CrossIcon />, deciding: undefined, disabled: undefined }
+
+export const ProposalStages = ({ roundStatus, updates, constitutionality, value, onChange }: ProposalStagesProps) => {
+
+  const rounds: RoundState[] = useMemo(() => {
+
     const decidingCount = updates.filter(({ status }) => status === 'deciding').length
+
     const lastUpdate = last(updates).status
-    const onGoing = lastUpdate === status
+    const onGoing = lastUpdate === roundStatus
     const approved = lastUpdate === 'gracing'
     const rejected = !onGoing && !approved
-    const isDeciding = onGoing && status === 'deciding'
-    const isDormant = onGoing && status === 'dormant'
-    return [
-      ...repeat((round) => ({ icon: <CheckboxIcon />, onClick: () => onChange(round) }), decidingCount - 1),
-      {
-        icon: !isDeciding && (rejected ? <CrossIcon /> : <CheckboxIcon />),
-        onClick: () => onChange(decidingCount - 1),
-      },
-      ...(isDormant ? [{ icon: false, onClick: undefined }] : []),
-    ]
-  }, [updates.length, status])
+    const isDormant = onGoing && roundStatus === 'dormant'
+
+    return repeat((round) => {
+      if (round < decidingCount) {
+        return `approved`
+      } else if (round > decidingCount) {
+        return `disabled`
+      } else if (isDormant || approved) {
+        return `approved`
+      } else if (rejected) {
+        return `rejected`
+      } else {
+        return `deciding`
+      }
+    }, constitutionality ?? 1)
+  }, [updates.length, roundStatus, constitutionality])
+
 
   return (
     <TabsContainer>
-      {rounds.map(({ icon, onClick }, round) => (
-        <TabContainer key={round} active={round === value} disabled={!onClick} onClick={onClick}>
-          {icon}
-          <span>
-            Council approvals {round + 1}/{constitutionality}
-          </span>
-        </TabContainer>
-      ))}
+      {rounds.map((roundStatus, round) => {
+        const isDisabled = roundStatus === 'disabled'
+        const isActive = round === value
+        const onClick = isDisabled ? undefined : () => { return onChange(round) }
+        const icon = iconMap[roundStatus]
+        return (
+
+          <Tooltip
+            key={round}
+            tooltipText={
+              !isDisabled
+                ? 'This proposal must undergo the voting of multiple consequent councils. The result of each council vote will be displayed in the separate tabs. For the proposal to be approved, each of the councils must approve it.'
+                : 'The number of councils in that must approve the proposal in a row before it has its intended effect is more than one. The overall execution of proposal will be triggered after the last council completes the voting. Outcomes of each council voting is displayed in a separate tab.'
```

Comment:
```suggestion
                : 'The number of councils that must approve the proposal in a row before it has its intended effect is more than one. The execution is triggered after the last council completes the voting. Votes for each council are displayed in a separate tab.'
```
Question: In above screenshot voting is complete but status is "Deciding". What am i missing?
> The execution is triggered after the last council completes the voting.

Reply:
@traumschule it shows 3 votes but they are the votes from the first round so it looks to me like it the issue #4201